cache/cache.go

```diff
@@ -27,3 +27,8 @@ type Item struct {
 	Value  interface{}
 	Expiry time.Time
 }
+
+// HasExpired returns true if the item has expired.
+func (i *Item) HasExpired() bool {
```

Comment:
Wonder if we should expose this as expired Items would never be returned right? Could keep it wrapped within `cache` to simplify the API and reduce change of misleading the client.